nautobot/core/views/__init__.py

```diff
@@ -221,6 +221,8 @@ def csrf_failure(request, reason="", template_name="403_csrf_failure.html"):
     return HttpResponseForbidden(t.render(context), content_type="text/html")
 
 
+# LoginRequiredMixin cannot be added here as this is both available to the UI and API
+# if added on UI a redirect tot login occurs which is good but on API it also redirects to Login page which is bad
```

Comment:
Also, redirecting the docs when not LoggedIn is not straight forward because Django does not really handle static files, and in production it may be managed by ngnix or something else.

Reply:
What would you like to do here? Heuristic on the `render_graphiql` to check to see if the browser has `accept: html`?

Reply:
`render_graphiql` should be UI-specific, I can't think that it would be used in the API?